pkg/collector/upgrade/v_0_122_0_test.go

```diff
@@ -0,0 +1,201 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package upgrade
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+
+	"github.com/open-telemetry/opentelemetry-operator/apis/v1beta1"
+)
+
+var logger = logf.Log.WithName("unit-tests")
+
+func TestUpgrade0_122_0(t *testing.T) {
+	tests := []struct {
+		name           string
+		initialConfig  *v1beta1.OpenTelemetryCollector
+		expectedConfig *v1beta1.OpenTelemetryCollector
+	}{
```

Comment:
can we add tests for the following:
* someone sets a prom reader already, we should do nothing
* someone sets sending otlp instead of prom, we should do nothing